sdk/storage/Microsoft.Azure.WebJobs.Extensions.Storage.Blobs/src/Listeners/SharedBlobQueueListenerFactory.cs

```diff
@@ -68,14 +67,32 @@ public SharedBlobQueueListener Create()
 
             // This special queue bypasses the QueueProcessorFactory - we don't want people to override this.
             // So we define our own custom queue processor factory for this listener
-           var queueProcessor = new SharedBlobQueueProcessor(triggerExecutor, _hostBlobTriggerQueue, defaultPoisonQueue, _loggerFactory, _queueOptions);
+            var queuesOptions = BlobsOptionsToQueuesOptions(_blobsOptions);
+            var queueProcessor = new SharedBlobQueueProcessor(triggerExecutor, _hostBlobTriggerQueue, defaultPoisonQueue, _loggerFactory, queuesOptions);
             QueueListener.RegisterSharedWatcherWithQueueProcessor(queueProcessor, _sharedQueueWatcher);
             IListener listener = new QueueListener(_hostBlobTriggerQueue, defaultPoisonQueue, triggerExecutor, _exceptionHandler, _loggerFactory,
-                _sharedQueueWatcher, _queueOptions, queueProcessor, _functionDescriptor, functionId: SharedBlobQueueListenerFunctionId);
+                _sharedQueueWatcher, queuesOptions, queueProcessor, _functionDescriptor, functionId: SharedBlobQueueListenerFunctionId);
 
             return new SharedBlobQueueListener(listener, triggerExecutor);
         }
 
+        internal static QueuesOptions BlobsOptionsToQueuesOptions(BlobsOptions blobsOptions)
+        {
+            // The maximum parallelism of QueueListener is BatchSize + NewBatchThreshold when configuring queue options. I.e. extension will keep requesting new batches until number
+            // of tasks that are still processing messages is below NewBatchThreshold.
+
+            // Split MaxDegreeOfParallelism between BatchSize and NewBatchThreshold. Cap at MaxBatchSize to not exceed the limit and make sure there's at least 1 message pulled
+            // if MaxDegreeOfParallelism is 1.
+            int batchSize = Math.Min(QueuesOptions.MaxBatchSize, blobsOptions.MaxDegreeOfParallelism / 2 + 1);
+            int newBatchThreshold = blobsOptions.MaxDegreeOfParallelism - batchSize;
```

Comment:
If a user sets max concurrency to 1, it looks like you'll translate this to BatchSize = 1 and NewBatchThreshold = 7 (on a single proc machine). So they won't get the serial behavior they were looking for?